### PR TITLE
Add x namespace for experimental libraries and inthashmap implementation

### PR DESCRIPTION
**Pull request by gabixdev**
`x` namespace was suggested by Alex during Discord talk. The `hashmap` library focuses on hashmap implementations optimized for specific data types eg. int-anything or usecases eg. synchronized/thread safe map.

to-dos:
- make the API between `IntHashMap` and `hashmap` consistent
- add language-level support for hashmaps and make it use a type-specific implementation if needed. eg.
```v
nummap := hashmap[int]int // creates a hashmap using integer hashmap implementation
strmap := hashmap[string]int // creates a hashmap using default hashmap implementation

nummap[1] = 5
nummap[3] = 6
nummap['test'] = 7 // won't compile
println(nummap[3]) // 6

strmap['abc'] = 5
strmap['def'] = 6
println(nummap['abc]) // 5
```

### CODE REVIEW

**Comment by gabixdev** [comment]
Tests fail due to a compiler bug, depends on #2210 to be fixed.

**Comment by dumblob** [comment]
I was unable to recognize which algorithm this `int` hashmap implements - could you please document it in comments?

Btw. did you take a look at the most recent existing [rigorous benchmark of many existing best performing hash tables](https://martin.ankerl.com/2019/04/01/hashmap-benchmarks-01-overview/ )? The current `int` hashmap doesn't look that sophisticated to be as performant as are the hashmaps in the linked test from Martin Ankerl.

Also when implementing any hashmap, please consider making it secure (it's one of the *V*'s goals if I'm not mistaken) by randomizing the iteration order - read carefully [this discussion](https://github.com/rust-lang/rust/issues/36481 ) about such attacks and possible [highly performant mitigations](https://github.com/rust-lang/rust/issues/36481#issuecomment-338910908 ).

@medvednikov is the `x` namespace meant to become something like `from future import ...` in Python or the *staging/* directory in vanilla Linux?

File: fns.txt [hunk 1]
```diff
@@ -0,0 +1,102 @@
+new_array
```
Comment (spytheman) [inline fns.txt]: fns.txt should be removed from the PR. It is a generated file by `v -debug`

Comment (medvednikov) [reply fns.txt]: yes, please remove fns.txt

**Pull request closed without merging**